demos/common/cpp/utils/include/utils/kuhn_munkres.hpp

```diff
@@ -31,23 +34,6 @@ class KuhnMunkres {
     std::vector<size_t> Solve(const cv::Mat &dissimilarity_matrix);
 
 private:
-    static constexpr int kStar = 1;
-    static constexpr int kPrime = 2;
-
-    cv::Mat dm_;
-    cv::Mat marked_;
-    std::vector<cv::Point> points_;
-
-    std::vector<int> is_row_visited_;
-    std::vector<int> is_col_visited_;
-
-    int n_;
-
-    void TrySimpleCase();
-    bool CheckIfOptimumIsFound();
-    cv::Point FindUncoveredMinValPos();
-    void UpdateDissimilarityMatrix(float val);
-    int FindInRow(int row, int what);
-    int FindInCol(int col, int what);
-    void Run();
+    class Impl;
+    std::shared_ptr<Impl> impl_;  ///< Class implementation.
```

Comment:
I don't think we need implementation in a separate class. Please move everything into actual `KuhnMunkres`.